project/src/main/java/com/google/sps/servlets/EventRegistrationServlet.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.cloud.Date;
+import com.google.gson.Gson;
+import com.google.sps.data.Event;
+import com.google.sps.data.User;
+import com.google.sps.utilities.CommonUtils;
+import com.google.sps.utilities.SpannerTasks;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/register-event")
+public class EventRegistrationServlet extends HttpServlet {
+  /** Update event with logged in user as attendee when they hit register */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String eventId = request.getParameter("eventId");
+    Optional<Event> eventOptional = SpannerTasks.getEventById(eventId);
+
+    // If event DNE, sends 404 ERR to frontend
+    if (eventOptional.isPresent()) {
```

Comment:
Consider exiting early here with the negative condition to make the precondition clear and the code below more width-efficient.

https://screenshot.googleplex.com/T0TrG0POLdp